modules/tensor_mechanics/src/actions/DynamicTensorMechanicsAction.C

```diff
@@ -0,0 +1,116 @@
+/****************************************************************/
```

Comment:
In this case we will have to refactor the TM action and break the individual pieces up into virtual member functions (one that sets up the parameters and adds the kernel for example). You can override them in the derived class.